src/ssl_cert_paths_discover.c

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2010-2021, Tarantool AUTHORS, please see AUTHORS file.
+ *
+ * Redistribution and use in source and binary forms, with or
+ * without modification, are permitted provided that the following
+ * conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above
+ *    copyright notice, this list of conditions and the
+ *    following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above
+ *    copyright notice, this list of conditions and the following
+ *    disclaimer in the documentation and/or other materials
+ *    provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY <COPYRIGHT HOLDER> ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * <COPYRIGHT HOLDER> OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+ * BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <dirent.h>
+#include <sys/stat.h>
+#include <openssl/ssl.h>
+#include "ssl_cert_paths_discover.h"
+#include "tt_static.h"
+
+static int
+is_dir_empty(const char *dir_path)
+{
+	DIR *dir = opendir(dir_path);
+	if (!dir)
+		return 1;
+
+	struct dirent *ent;
+	int is_empty = 1;
+	while ((ent = readdir(dir))) {
+		if (strcmp(ent->d_name, ".") && strcmp(ent->d_name, "..")) {
+			if (ent->d_type == DT_REG ||
+			    ent->d_type == DT_LNK ||
+			    ent->d_type == DT_DIR) {
+				is_empty = 0;
+				break;
+			}
+		}
+	}
+
+	closedir(dir);
+	return is_empty;
+}
+
+const char *DEFAULT_CERT_DIR_PATHS[] = {
+	/* Fedora/RHEL/Centos */
+	"/etc/pki/tls/certs",
+	/* Debian/Ubuntu/Gentoo etc. (OpenSuse) */
+	"/etc/ssl/certs",
+	/* FreeBSD */
+	"/usr/local/share/certs",
+	/* NetBSD */
+	"/etc/openssl/certs",
+	/* macOS */
+	"/private/etc/ssl/certs",
+	"/usr/local/etc/openssl@1.1/certs",
+	"/usr/local/etc/openssl@1.0/certs",
```

Comment:
I see some downsides of hardcoding these paths:
1. Imagine you have your own ca certificate that you cannot place to a system file. How you would use it with Tarantool?
2. For each release we will need to review these paths and update for new operating systems releases and remove old ones.

Have you reviewed approaches in other opensource projects?
I found source code and documentation in curl and Fossil SCM that solves similar problems:
Fossil SCM [1] (they use environment variables) and curl that allows to specify a path to file with CA certificates. Also curl allows to ignore certificate chain check and make connection insecure.

1. https://github.com/curl/curl/tree/master/lib/vtls
2. https://curl.se/docs/sslcerts.html
3. https://www.fossil-scm.org/home/file?name=src/http_ssl.c&ci=trunk

Reply:
About hardcoing paths (this approach used in golang and kafka - see previous answer):
1. User can set SSL env variables matching his situation. They won't be overwritten since function `ssl_cert_paths_discover` is called with `overwrite` = 0 param. 

2. This may be so, but as I researched these paths don't change frequently.

I've reviewd other approaches (see previous answer about golang and librdkafka), also nodejs use hardcoded cert.pem as
string at source code. Here are a links:
- definition of cert: https://github.com/nodejs/node/blob/44a66adbaa9d82cf70b6569701359b9a871acd31/src/node_root_certs.h
- cert using: https://github.com/nodejs/node/blob/f5287a4b7b0b14f69b264d8017b76c100f5bc2c1/src/crypto/crypto_context.cc#L42


Also approach used in this patch still allows user to turn off ssl verification for ex. by setting SSL env vars to empty (and call http.request with verify_peer = false).